CMakeLists.txt

```diff
@@ -98,8 +130,19 @@ elseif(BUILD_COMPILER STREQUAL "msvc")
     add_compile_options(/MP) # Multi-threaded build
     add_compile_options(/Zc:preprocessor) # Use standard compliant preprocessor
     add_compile_options(/we4834) # Discarding return value of function with 'nodiscard' attribute
-    add_link_options(/LARGEADDRESSAWARE) # Enable heap size over 2gb for x86 builds
-    add_link_options(/SAFESEH:NO) # This is for x86 build
+    if (CHECK_CPU_ARCHITECTURE_X86)
```

Comment:
`CHECK_CPU_ARCHITECTURE_X86` comes from SDL internals, let's refrain from using it. `Detection.cmake` defines `BUILD_TYPE` to `"x86"`, so this should be `BUILD_TYPE STREQUAL "x86"`